vsintegration/src/vs/FsPkgs/FSharp.Project/FS/MSBuildUtilities.fs

```diff
@@ -12,6 +12,60 @@ open System.Diagnostics
 open System.IO
 open System
 
+module internal MSBuildHelpers =
+
+    let FolderComparer = StringComparer.OrdinalIgnoreCase 
+
+    let GetPathParts (x:string) =
+        x.Split( [| '\\' |], StringSplitOptions.RemoveEmptyEntries) |> Seq.ofArray
+
+    let IsSameDir a b =
+        0 = Seq.compareWith (fun x y -> FolderComparer.Compare(x,y)) (GetPathParts a) (GetPathParts b)
+
+    let IsSubDirOf baseDir directory  =
+        let rec helper b d = 
+            match b, d with
+            | [], [] -> false
+            | [], _ -> true
+            | _, [] -> false
```

Comment:
This case is not needed I don't think.

Could flatten this even more if you want:
```fsharp
let rec helper b d = 
    match b, d with
    | [], _ :: _ -> true
    | x :: xs, y :: ys when FolderComparer.Compare(x,y) = 0  -> helper xs ys
    | _ -> false
```